src/app/trade-creator-page/trade-creator-preview/trade-creator-preview.component.ts

```diff
@@ -121,8 +123,30 @@ export class TradeCreatorPreviewComponent implements OnInit {
           this.creatorCoinTrade.expectedCreatorCoinReturnedNanos = ExpectedCreatorCoinReturnedNanos || 0;
           this.creatorCoinTrade.expectedBitCloutReturnedNanos = ExpectedBitCloutReturnedNanos || 0;
 
-          // This will update the user's balance.
-          this.appData.updateEverything(response.TxnHashHex, this._creatorCoinSuccess, this._creatorCoinFailure, this);
+          if (
+            this.creatorCoinTrade.followCreator &&
+            !this.followService._isLoggedInUserFollowing(this.creatorCoinTrade.creatorProfile.PublicKeyBase58Check)
+          ) {
+            this.followService
+              ._toggleFollow(true, this.creatorCoinTrade.creatorProfile.PublicKeyBase58Check)
+              .add(() => {
+                // This will update the user's balance.
+                this.appData.updateEverything(
+                  response.TxnHashHex,
+                  this._creatorCoinSuccess,
+                  this._creatorCoinFailure,
+                  this
+                );
+              });
+          } else {
+            // This will update the user's balance.
+            this.appData.updateEverything(
+              response.TxnHashHex,
+              this._creatorCoinSuccess,
+              this._creatorCoinFailure,
+              this
+            );
+          }
```

Comment:
we may be able to clean this up a bit (note: `import { of } from "rxjs";`  at the top of this file):
```
const observable = this.creatorCoinTrade.followCreator &&
            !this.followService._isLoggedInUserFollowing(this.creatorCoinTrade.creatorProfile.PublicKeyBase58Check) ? this.followService._toggleFollow(true, this.creatorCoinTrade.creatorProfile.PublicKeyBase58Check) : of(null);
observable.add(() => {
this.appData.updateEverything(
              response.TxnHashHex,
              this._creatorCoinSuccess,
              this._creatorCoinFailure,
              this
});
```